### DISCUSSION TITLE
[Question] CarouselView NextItem and PreviousItem

### DISCUSSION THREAD

**Discussion by AlleSchonWeg** [xamarin/Xamarin.Forms#11442] [General]
Hi,
i need to access the content from the nextItem and the previousItem in a CarouselView. Is it possible? I have a ScrollView as a child of a datatemplate and if the user scrolls in the currentItem i want to sync the scroll position in nextItem and previousItem. Because if the user swipes to another item the scroll position must be the same.
```
                      <CarouselView.ItemTemplate>
						<DataTemplate>
							<StackLayout Orientation="Vertical">
								<ScrollView BackgroundColor="Red"
											x:Name="scrollView"
											Scrolled="ScrollView_Scrolled">
									...Scrollable data...
								</ScrollView>
							</StackLayout>
						</DataTemplate>
					</CarouselView.ItemTemplate>
```
@jsuarezruiz Sorry for tagging, but i think you are the expert :)

**Comment by davidortinau** [comment]
If you want the `ScrollView`s in previous, current, and next items (assuming they all exist), then I would recommend using a customer scroll view or a behavior on the `ScrollView` and **binding** the position to a `ViewModel` property that is shared among them all. At least that's where I would start. 

Here's a custom `ScrollView` I've made that exposes the scroll position.

https://github.com/davidortinau/Xappy/blob/main/Xappy/Xappy/Content/Common/ScrollerView.cs

**Reply by AlleSchonWeg** [reply]
Hi @davidortinau 
Sorry for me late reply.
Creating a custom ScrollView and binding the position to a ViewModel will not work if the control is a children of datatemplate:
```
<CarouselView.ItemTemplate>
	<DataTemplate>
      <ctrl:CustomScrollView BackgroundColor="Red"
                             PositionY="{Binding ScrollYPosition, Mode=TwoWay }"
							 x:Name="scrollView">
									...Scrollable data...
      </ctrl:CustomScrollView >
	</DataTemplate>
 </CarouselView.ItemTemplate>
```

If you update the ScrollYPosition Value(somewhere in code behind) the propertyChanged event(s) of the BindableProperty PositionY for the custom ScrollViews are not fired. So you can't share the binding among them all.
If you move the custom scrollview declaration outside the ItemTemplate binding works as expected.
